### PR TITLE
Refactoring & minor improvements

### LINKED ISSUE #28: `compile()` to specify source filename in `creepy.pipe.connect()` loader
[compile](https://docs.python.org/3/library/functions.html#compile)

### PR DESCRIPTION
**Pull request by progbeat**
Closes #28

### CODE REVIEW

File: creepy/_detail/private_key.py [hunk 1]
```diff
@@ -23,7 +23,8 @@ def _get_private_key():
 
 
 def load(path: str, passphrase):
-    with creepy.pipe.connect('private_numbers') as session:
+    with creepy.pipe.connect('private_numbers',
+                             hash='5d23c98f71924042eb30dfae4d4a8a7d97edba75c3b9cc98fd9293373833d8dd') as session:
```
Comment (romawhite47) [inline creepy/_detail/private_key.py:27]: Does this mean that each time you change `private_numpers.py` you should change this hash?

Comment (progbeat) [reply creepy/_detail/private_key.py:27]: Yes.

Comment (progbeat) [reply creepy/_detail/private_key.py:27]: For debugging you can omit hash.

Comment (romawhite47) [reply creepy/_detail/private_key.py:27]: So maybe add such comment to `private_numbers.py` or test that checks hashes.

Comment (progbeat) [reply creepy/_detail/private_key.py:27]: I'm not sure I understand what you mean. I specify hash to make sure I run genuine 'private_numbers', 'private_numbers' itself doesn't know it runs with hash verification or without. Making test that check hashes is pointless as it doesn't make sure 'private_numbers' is genuine.

Comment (romawhite47) [reply creepy/_detail/private_key.py:27]: I mean that if I add whitespace to private_number then some other part of code stop working. It's not what you usually expect. User of this lib should know that he shouldn't make such changes. One option is to make test so he can run pytest and see that he shouldn't change it.

Comment (progbeat) [reply creepy/_detail/private_key.py:27]: It's intended documented behaviour. User should not touch `private_numbers` unless he knows what he does. If he changed it by mistake or it was changed by attacker then `connect()` will raise exception telling that script isn't genuine, so it won't run it. Then user should revert changes or notice that script was hacked. If user changed script intentionally, he will copy-paste hash from exception message.

**Review by romawhite47 — APPROVED** [review]
LGTM

**Pull request merged**